tests/ui/indexing_slicing.rs

```diff
@@ -1,18 +1,27 @@
 #![feature(plugin)]
-
-
 #![warn(indexing_slicing)]
 #![warn(out_of_bounds_indexing)]
 #![allow(no_effect, unnecessary_operation)]
 
 fn main() {
-    let x = [1,2,3,4];
+    let x = [1, 2, 3, 4];
+    let index: usize = 1;
+    let index_from: usize = 2;
+    let index_to: usize = 3;
+    x[index];
+    &x[index_from..index_to];
+    &x[index_from..][..index_to];
+    &x[index..];
+    &x[..index];
     x[0];
     x[3];
     x[4];
     x[1 << 3];
     &x[1..5];
+    &x[1..][..5];
     &x[0..3];
+    &x[0..][..3];
+    &x[0..].get(..3); // Ok
```

Comment:
Wait this should trigger the `INDEXING_SLICING` lint too (on `x[0..]`). Or am I missing something? What happens if you do something like `&x[5..].iter().map(|x| 2 * x).collect::<Vec<i32>>()`? This should definitely trigger the lint. Could you add this test with a range `5..` (panic) and a range `2..` (no panic). This got linted before the change.

Reply:
Oh sorry for the confusion! `x[0..]` __does not__ trigger the lint and was put in this test file to ensure that stderr was not generated for it. And I will add the example you mentioned :+1:

Reply:
Ahhh I misunderstood this then, thanks for the clarification!